File: point-service/src/main/java/com/store/point/consumer/SignupEventConsumer.java
package com.store.point.consumer;

import com.store.common.event.SignupEvent;
import com.store.point.domain.PointSaveDomain;
import com.store.point.service.PointService;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.kafka.annotation.KafkaListener;
import org.springframework.stereotype.Component;

@Slf4j
@Component
@RequiredArgsConstructor
public class SignupEventConsumer {

    private final PointService pointService;

    @KafkaListener(topics = "user.signup.point", groupId = "point-group")
    public void consume(SignupEvent signupEvent) {
        // 회원가입 성공!
        log.info("signup success: {}", signupEvent.toString());
        PointSaveDomain pointSaveDomain = new PointSaveDomain(signupEvent.id(), 1000L, "회원가입 시 지급되는 포인트", "signup");
        pointService.savePoint(pointSaveDomain);
    }
}

File: notification-service/src/main/java/com/store/notification/infrastructure/kafka/SignupEventConsumer.kt
package com.store.notification.infrastructure.kafka

import com.store.common.event.SignupEvent
import com.store.notification.domain.notification.service.NotificationService
import org.slf4j.LoggerFactory
import org.springframework.kafka.annotation.KafkaListener
import org.springframework.stereotype.Component

@Component
class SignupEventConsumer(
    private val notificationService: NotificationService
) {

    private val log = LoggerFactory.getLogger(this::class.java)

    @KafkaListener(topics = ["user.signup.mail"], groupId = "notification-group")
    fun consume(event: SignupEvent) {
        try {
            log.info("kafka message received $event")
            //notificationService.sendSignupEmail(event)
        } catch (e: Exception) {
            log.error("Failed to process message: ${e.stackTraceToString()}")
        }
    }
}
